blockchain/pallets/minting/src/tests.rs

```diff
@@ -282,6 +287,63 @@ mod register_identity {
         });
     }
 
+    #[test]
+    fn first_call_to_register_for_minting_is_free() {
+        use frame_support::pallet_prelude::Pays;
+
+        new_test_ext().execute_with(|| {
+            register_id_account(42, 1);
+
+            // first call is free, actual_weight set to 0
+            let tree_0 = gen_tree(&["a", "b"]);
+            assert_eq!(
+                FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_0),
+                Ok(PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::No
+                })
+            );
+
+            // second is charged, actual_weith None indicates annotation
+            // weight is applied
+            let tree_1 = gen_tree(&["a", "b", "c"]);
+            assert_eq!(
+                FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_1),
+                Ok(PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::Yes
+                })
+            );
```

Comment:
I would add this as a separate test. The description of this test says the first minting is free, and this doesn't exactly relate to that. If you're doing work after assertions in a test, that's usually a sign you should have two tests.

Reply:
I agree, I have split this up into 2 tests, where both only have on assert at the end